lib/pure/strutils.nim

```diff
@@ -367,15 +367,20 @@ iterator split*(s: string, seps: set[char] = Whitespace): string =
   ##   "08.398990"
   ##
   var last = 0
+  var splits = maxsplit
   assert(not ('\0' in seps))
   while last < len(s):
     while s[last] in seps: inc(last)
     var first = last
     while last < len(s) and s[last] notin seps: inc(last) # BUGFIX!
     if first <= last-1:
+      if splits == 0:
+        yield substr(s, first, len(s)-1)
```

Comment:
AFAIK, every yield inside an inline iterator causes a copy of the `for` code and thus extra yields have to be avoided.